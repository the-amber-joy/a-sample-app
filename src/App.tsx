import { ChakraProvider, theme } from "@chakra-ui/react";
import { Layout } from "./components/layout/Layout";

export const App = () => {
  return (
    <ChakraProvider theme={theme}>
      <Layout />
    </ChakraProvider>
  );
};
