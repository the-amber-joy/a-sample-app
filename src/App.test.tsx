import { screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { render } from "./test-utils";
import { App } from "./App";

test("renders Loading Pokemon text initially", () => {
  render(<App />);
  const el = screen.getByText("Loading Pokemon");
  expect(el).toBeInTheDocument();
});
