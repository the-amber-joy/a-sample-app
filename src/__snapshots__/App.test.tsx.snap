// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <div
    className="css-12hvf5p"
  >
    <div
      className="css-1uwjw6e"
    >
      <button
        aria-label="Switch to dark mode"
        className="chakra-button css-rpznon"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          fill="currentColor"
          focusable={false}
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M283.211 512c78.962 0 151.079-35.925 198.857-94.792 7.068-8.708-.639-21.43-11.562-19.35-124.203 23.654-238.262-71.576-238.262-196.954 0-72.222 38.662-138.635 101.498-174.394 9.686-5.512 7.25-20.197-3.756-22.23A258.156 258.156 0 0 0 283.211 0c-141.309 0-256 114.511-256 256 0 141.309 114.511 256 256 256z"
          />
        </svg>
      </button>
    </div>
    <div
      className="css-sqchgo"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="chakra-form-control css-1kxonj9"
          role="group"
        >
          <div
            className="chakra-stack css-pok3cj"
          >
            <input
              className="chakra-input css-f21rl8"
              disabled={false}
              id="field-:r0:"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Search by name"
              readOnly={false}
              required={false}
              value=""
            />
            <button
              aria-label="Search by name"
              className="chakra-button css-22xpxh"
              onClick={[Function]}
              type="submit"
            >
              <svg
                aria-hidden={true}
                className="chakra-icon css-onkibi"
                focusable={false}
                viewBox="0 0 24 24"
              >
                <path
                  d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
        </div>
      </form>
      <div
        className="chakra-card css-1s44or5"
      >
        <div
          className="chakra-card__header css-1sl53ol"
        >
          <h2
            className="chakra-heading css-18j379d"
          >
            Loading 
            Pokemon
          </h2>
        </div>
        <div
          className="chakra-card__body css-1idwstw"
        >
          <div
            className="css-gmuwbf"
          >
            <div
              className="chakra-spinner css-97v01s"
            >
              <span
                className="css-8b45rq"
              >
                Loading...
              </span>
            </div>
          </div>
        </div>
        <hr
          aria-orientation="horizontal"
          className="chakra-divider css-svjswr"
        />
        <div
          className="chakra-card__footer css-yqdxxt"
        >
          <div
            className="chakra-button__group css-rl82lu"
            data-orientation="horizontal"
            role="group"
          >
            <button
              className="chakra-button css-4nu9hl"
              onClick={[Function]}
              type="button"
            >
              Pick Another!
            </button>
            <button
              className="chakra-button css-1qpdsri"
              onClick={[Function]}
              type="button"
            >
              Make it Shiny!
            </button>
          </div>
        </div>
      </div>
      <div
        className="chakra-stack css-khgzdf"
      >
        <h2
          className="chakra-heading css-1dklj6k"
        >
          Favorites
        </h2>
        <div
          className="chakra-stack css-1iz79u8"
        />
      </div>
    </div>
  </div>,
  <span />,
  <span
    hidden={true}
    id="__chakra_env"
  />,
]
`;
